bot/exts/filtering/_ui/ui.py

```diff
@@ -681,12 +685,14 @@ def _extract_potential_phish(
             if len(content_list) > 1:
                 return "", None
             if content_list:
-                content = next(iter(content_list))
-                if filter_list.name == "domain" and "discord" in content:  # Leave invites to the invite filterlist.
+                current_content = next(iter(content_list))
+                if filter_list.name == "domain" and re.fullmatch(DISCORD_INVITE, current_content):
+                    # Leave invites to the invite filterlist.
                     continue
                 if encountered:
                     return "", None
                 target_filter_list = filter_list
+                content = current_content
                 encountered = True
```

Comment:
The former.

This change fixe the first bug I've mentioned in the PR description.
Invite links of the form `https://discord.gg/aBcDeFgH`  will be present in `content_list` twice. Once as an invite filter and once as a domain filter (..and in this order from my testing. This is why we `continue` to the next iteration of the loop if it's domain filter and the content of the filter matches the pattern of a guild invite.

Before this change, at the point where we encounter `continue`, `content` will be the lowercased version of the link. If we are at the last element of `content_list` we return this lowercased link when we should have actually reverted to the previous value of `content` when we hit the `continue` statement.


Also not directly related to your question: `encountered` is kind of redundant now because we can now simply check if `content` truthy- but I've left it unchanged for code clarity.

Reply:
The former.

This change fixe the first bug I've mentioned in the PR description.
Invite links of the form `https://discord.gg/aBcDeFgH`  will be present in `content_list` twice. Once as an invite filter and once as a domain filter (..and in this order from my testing. This is why we `continue` to the next iteration of the loop if it's domain filter and the content of the filter matches the pattern of a guild invite.

Before this change, at the point where we encounter `continue`, `content` will be the lowercased version of the link. If we are at the last element of `content_list` we return this lowercased link when we should have actually reverted to the previous value of `content` when we hit the `continue` statement.


Also not directly related to your question: `encountered` is kind of redundant now because we can now simply check if `content` truthy- but I've left it unchanged for code clarity.

Reply:
Right, right. Thanks for explaining, that makes sense.